connPFM/utils/io.py

```diff
@@ -0,0 +1,54 @@
+import logging
+import subprocess
+
+import numpy as np
+from nilearn.input_data import NiftiLabelsMasker
+
+from connPFM.utils import atlas_mod
+
+LGR = logging.getLogger(__name__)
+
+
+def load_data(data, atlas, n_echos=1):
+    """
+    Load and mask data with atlas using NiftiLabelsMasker.
+    """
+    # Initialize masker object
+    masker = NiftiLabelsMasker(labels_img=atlas, standardize=False, strategy="mean")
+
+    # If n_echos is 1 (single echo), mask and return data
+    if n_echos == 1:
+        # If data is a list, keep only first element
+        if isinstance(data, list):
+            data = data[0]
+        data_masked = masker.fit_transform(data)
+    else:
+        # If n_echos is > 1 (multi-echo), mask each echo in data list separately and
+        # concatenate the masked data.
+        for echo_idx, echo in enumerate(data):
+            if echo_idx == 0:
+                data_masked = masker.fit_transform(echo)
+            else:
+                data_masked = np.concatenate((data_masked, masker.fit_transform(echo)), axis=0)
+
+    return data_masked, masker
+
+
+def save_img(data, output, masker, history_str=None):
```

Comment:
So your output image right now is `nscans*nechos x nvoxels` in 1 nifti image?

Reply:
No. The data that goes into `save_img()` will always be `nscans x nvoxels`.

However, you have reminded me we have to export the fitted signal in a for loop. That should be the only thing missing on this PR.

Reply:
Ah okay you are doing the for loop outside

Reply:
That's right. I think this is the cleanest way of doing it.